sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1682,6 +1682,22 @@ object SQLConf {
     .doubleConf
     .createWithDefault(0.9)
 
+  val OPTIMIZE_INTERSECT_ENABLED =
+    buildConf("spark.sql.cbo.optimizeIntersect.enabled")
+      .internal()
+      .doc("Whether to use optimized Intersect implementation or not. " +
+        "Optimized Intersect logic tries to pushdown Distinct through Join based on some stats")
+      .booleanConf
+      .createWithDefault(true)
+
+  val OPTIMIZE_INTERSECT_DISTINCT_REDUCTION_THRESHOLD =
+    buildConf("spark.sql.cbo.optimizeIntersect.distinctReductionThreshold")
+      .internal()
+      .doc("Ratio by which Distinct should reduce number of rows to qualify for pushdown" +
+        " in Intersect optimization")
+      .intConf
+      .createWithDefault(100)
```

Comment:
How about merging the two options into one by indicating `-1` to deactivate the rule?

Reply:
@maropu We also want to control whether pushdown should happen when right side is broadcastable.

Reply:
Rather, we need an new option to turn on/off this rule? Any negative impact on performance?

Reply:
No I didn't find any performance regression with this stats based triggers. I kept everything behind config based on other configs used in similar features (Ex - costBasedjoinReorder).

Should we remove the "spark.sql.cbo.optimizeIntersect.enabled" config here.

Reply:
Yea, could you remove it now? If we need to revert it back in the result of following reviews, I think that it should be added again at that time.

btw, I think this is not a feature of cbo, so you don't need to add new configs under the cbo namespace. We already have some optimizer rules depending on stats, e.g., `spark.sql.optimizer.dynamicPartitionPruning.useStats`.

Reply:
@maropu I have removed cbo namespace from the config names.

But I still think that we should have a config flag for this stats based optimization - same way other stats based optimization has Ex - costBasedJoinReordering, dynamicPartitionPruining.

Reply:
@prakharjain09 Hello, if the stats are wrong or incomplete (like the cardinality changed drastically after that last time we performed the analyze on the table) , we could possibly slow down especially when we push the distinct to the right leg , no ? @maropu isn't it better to keep this under a flag ?

Reply:
Yea, if we have the negative impact.